### DISCUSSION TITLE
Unknown Devices Appearing

### DISCUSSION THREAD

**Discussion by maleman82** [Ylianst/MeshCentral#4147] [General]
I discovered this morning that I have some strange unknown devices showing up in my Meshcentral in one specific group. The device names are "610930", "675052", and "art-PC". In my event logs I can the devices are being added to the group last night but there is NO user name associated with the entry. I also see this entry "[existing device name]→ Changed device [existing device name] from group [existing group name]: host"

Server version 1.0.12

Is my client hacked? Is my meshcentral hacked?

Please advise!
Thanks.

**Comment by fsfc-sysadmin** [comment]
Hi , i have the same issue .

art-PC is also apearing on one of my device groups.

**Comment by si458** [comment]
Could you possibly update your install as its quite old, the latest version is 1.0.60
Also delete the computers
Then change your password and see what happens?

**Comment by silversword411** [comment]
Are you suffering from thieving AV clients?

https://docs.tacticalrmm.com/faq/#help-ive-been-hacked-there-are-weird-agents-appearing-in-my-tactical-rmm

**Reply by fsfc-sysadmin** [reply]
I think this is the cause . Since we are using cloud based AV . They will probably download and run the installer in their sandbox , so it will appear to the meshcentral portal and never to be seen online again.

**Reply by si458** [reply]
Thinking about it now, I had the same weird thing happen when I was trying to get our certificate working with Norton,
I was uploading the exe to virustotal and every so often a random computer would appear,
I had a habit of connecting to the computer and using the Edge browser for a bit of free web surfing 🤣

**Comment by Ylianst** [comment]
Your server was likely NOT hacked. Sorry for not answering earlier.

Seeing strange computers show up on your server is (very oddly) normal and ok. Anti-virus companies will notice a new executable on your computer that they have not seen before and will download it and run it on their VM's to see what it does. This is common practice for them. Because your server information is embedded into the agent executable, the agent will connect back to your server from their test VM. It will stay for a few minutes and never show up again.

It's been reported many times on GitHub and Reddit. I don't think there is anything I can do about this, it's the anti-virus people doing their job.

Also note that someone installing an agent grants you access and rights to their machine, but they get to access to your server. So, even if someone pad installs the agent, it should not be a big deal... except that it can be annoying to see new devices show up.

Hope that makes sense.

**Reply by silversword411** [reply]
Need to start building a faq.

Got some screenshots of your av agents?

**Comment by silversword411** [comment]
https://github.com/Ylianst/MeshCentral/commit/fda71bf1e7661d2b06621e5286a0f7f4420fd50e

**Comment by maleman82** [comment]
Thanks everyone for your input! @Ylianst Thank you very much for weighing in on this and for your explanation. I have had interesting and other "odd" behavior before that was caused by AV's. So, while I don't like it, I can at least know what is going on with the machines.
Thanks again!

**Comment by Ylianst** [comment]
If there is every a suggest on what to do about this, do let me know. I don't think there is a good way to detect that a agent connection is from an AV vendor, but I am open to suggestions. I agree that it's annoying, but I don't mind AV vendors doing their job too.

**Comment by silversword411** [comment]
The only answer I have is setup whitelisting of devices by source IP (somehow) and only allow them access to your infra.

It's a lot of work, and impractical for public services like MeshCentral.com

**Comment by frakman1** [comment]
> It's been reported many times on GitHub and Reddit. I don't think there is anything I can do about this, it's the anti-virus people doing their job.

I have an idea. Maybe make the mesh agent executable only work within a short timeframe (few minutes or hours) so that a file that's hanging around on your file system can't be executed elsewhere and make a connection to the server. Either that or enforce the token validation at agent install time like the meshcentral website login and one of the command line tool does (`meshcmd`).

Having said that, there's no real danger here. The worst that can happen is that somebody else's machine is now controllable by _you_! That person does not have access to your meshcentral server if you have MFA enabled.
